eclipse-theia/theia
# Add Pinned Tabs

#### What it does

Commands for pinning and unpinning tabs. Fixes #9326. Closes #9614.

This is based on #9614 with a bug fix and a switch to codicons for pinned tab icons.

#### How to test

1. Use the tabbar context menu or command palette to pin and unpin tabs. 
   > In the main or bottom areas, they should receive a pin icon.
   If dirty, an editor should receive pin-with-dot icon.
2. Try to close pinned editors. They should not close, but should convert to unpinned editors.
3. Reload the page / restart the application. Pinned state should be restored.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
